File: src/main/java/com/libraryjava/controller/BookController.java
package com.libraryjava.controller;

import com.libraryjava.dto.book.BookLoanDto;
import com.libraryjava.dto.book.BookMakeDto;
import com.libraryjava.dto.book.BookReturnDto;
import com.libraryjava.dto.book.response.BookStatResponse;
import com.libraryjava.service.BookService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class BookController {
    private final BookService bookService;

    @PostMapping("/book")
    public ResponseEntity<Void> makeBook(@RequestBody BookMakeDto bookMakeDto) {
        bookService.makeBook(bookMakeDto);
        return ResponseEntity
                .ok().build();
    }

    @PostMapping("/book/loan")
    public ResponseEntity<Void> loanBook(@RequestBody BookLoanDto bookLoanDto) {
        try {
            bookService.loanBook(bookLoanDto);
            return ResponseEntity
                    .ok().build();
        } catch (IllegalArgumentException e) {
            return ResponseEntity
                    .badRequest().build();
        }
    }

    @PutMapping("/book/return")
    public ResponseEntity<Void> returnBook(@RequestBody BookReturnDto bookReturnDto) {
        bookService.returnBook(bookReturnDto);
        return ResponseEntity
                .ok().build();
    }

    @GetMapping("/book/loan")
    public ResponseEntity<Integer> getCountLoanedBook() {
        return ResponseEntity
                .ok(bookService.countLoanedBook());
    }

    @GetMapping("/book/stat")
    public ResponseEntity<List<BookStatResponse>> getBookStatistics() {
        return ResponseEntity
                .ok(bookService.getBookStatistics());
    }
}

File: library/src/main/kotlin/com/library/controller/book/BookController.kt
package com.library.controller.book

import com.library.dto.book.BookLoanRequest
import com.library.dto.book.BookRequest
import com.library.dto.book.BookReturnRequest
import com.library.dto.book.BookStatResponse
import com.library.service.book.BookService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class BookController(
    private val bookService: BookService
) {
    @PostMapping("/book")
    fun saveBook(@RequestBody request: BookRequest) {
        bookService.saveBook(request)
    }

    @PostMapping("/book/loan")
    fun loanBook(@RequestBody request: BookLoanRequest) {
        bookService.loanBook(request)
    }

    @PutMapping("/book/return")
    fun returnBook(@RequestBody request: BookReturnRequest) {
        bookService.returnBook(request)
    }

    @GetMapping("/book/loan")
    fun countLoanedBook(): Int {
        return bookService.countLoanedBook()
    }

    // 분야별로 등록되어 있는 책의 권수 보여주기
    @GetMapping("/book/stat")
    fun getBookStatistics() : List<BookStatResponse> {
        return bookService.getBookStatistics()
    }
}